### PR TITLE
ENH: functions filter_columns and filter_rows created

### LINKED ISSUE #55289: ENH: Improve Filter function with Filter_Columns and Filter_Rows
### Feature Type

- [X] Adding new functionality to pandas

- [ ] Changing existing functionality in pandas

- [ ] Removing existing functionality in pandas

### Problem Description

Using the [ ] syntax can get messy/complicated for applying filters to a dataframe. The current Filter() function is also confusing to use. I Propose adding dedicated functions to quickly filter out columns and rows to make Pandas easier to use. 

### Feature Description

Propose adding 2 new functions

Filter_Columns(), Filter_Rows()

def Filter_Columns( columns: List, inverse:Bool, inplace:Bool )
def Filter_Rows( Rows: List, inverse:Bool, inplace:Bool )

Usage: 

Filter_Columns( ['Names', 'Ages'  ],  inverse=False, inplace=True)   
# Shows columns for name and age.. Inverse is used to hide show other columns that are not name and age.

def Filter_Rows( [ ('name'==bob),  ( 'age' > 20)  ] , inverse:Bool, inplace:Bool )
# Shows dataframe where the value in names column =bob, and age column >20

Chained

Filter_Columns( ['Names', 'Ages'  ],  inverse=False, inplace=True).Filter_Rows( [ ('name'==bob),  ( 'age' > 20)  ] , inverse:Bool, inplace:Bool )

### Alternative Solutions

Use [ ] syntex... More confusing   

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by Cappuchinoo**
- [ ] closes #55289
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

Basically those 2 functions do the same as filter() in pandas/core/generic.py, but separated to deal with either columns or rows. Also hopefully it's easier without the need to specify the 'axis' and with the 'items' variable being replaced by 'colnames' and 'rownames'.

Is this close to what you expected @speed650? Your idea to have a list of boolean expressions (Filter_Rows( [ ('name'==bob), ( 'age' > 20) ])) be accepted by the functions sounded great, but I couldn't figure out how to do it without making them hard to use, defeating the purpose I suppose.

### CODE REVIEW

**Review by phofl — CHANGES_REQUESTED** [review]
Thx for your contribution, this needs a bit more discussion first though

**Comment by mroeschke** [comment]
Thanks for the PR but as mentioned consensus is needed in the issue before proceeding with a PR so closing for now but happy to reopen if there's agreement to move forward here

**Comment by Cappuchinoo** [comment]
Ok, I'll be checking from time to time to see if a consensus has been reached, thank you.
